### PR TITLE
Feature: meson toolchain

### PR DESCRIPTION
**Pull request by SSE4**
docs for https://github.com/conan-io/conan/pull/7662

### CODE REVIEW

**Review by czoido — CHANGES_REQUESTED** [review]
I think this docs should also be updated to the generate() method, here is the PR for the docs for the other toolchains: https://github.com/conan-io/docs/pull/1940/files

**Comment by SSE4** [comment]
> I think this docs should also be updated to the generate() method and new imports, here is the PR for the docs for the other toolchains: https://github.com/conan-io/docs/pull/1940/files

that probably will go to another PR. https://github.com/conan-io/conan/pull/7662 only added method `write_toolchain_files`. we will need to rename it in another PR and update docs.

**Comment by czoido** [comment]
I think that was already updated by @memsharded in conan-io/conan#8101 and conan-io/conan#8096, please check.

**Review by danimtb — APPROVED** [review]

File: creating_packages/toolchains/meson.rst [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    from conans import ConanFile
+    from conan.tools.meson import MesonToolchain
+
+    class App(ConanFile):
+        settings = "os", "arch", "compiler", "build_type"
+        requires = "hello/0.1"
+        options = {"shared": [True, False]}
+        default_options = {"shared": False}
+
+        def toolchain(self):
```
Comment (czoido) [inline creating_packages/toolchains/meson.rst:25]:
```suggestion
        def generate(self):
```
This should be supported by this toolchain

File: creating_packages/toolchains/meson.rst [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+MesonToolchain
+==============
+
+.. warning::
+
+    This is an **experimental** feature subject to breaking changes in future releases.
+
+
+The ``MesonToolchain`` can be used in the ``toolchain()`` method:
```
Comment (czoido) [inline creating_packages/toolchains/meson.rst:11]:
```suggestion
The ``MesonToolchain`` can be used in the ``generate()`` method:
```

File: creating_packages/toolchains/meson.rst [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            tc.generate()
+
+
+The ``MesonToolchain`` will generate the following file during ``conan install``
+command (or before calling the ``build()`` method when the package is being
+built in the cache): ``conan_meson_native.ini``
+
+``conan_meson_native.ini`` will contain the definitions of all the Meson properties
+related to the Conan options and settings for the current package, platform,
+etc. This includes but is not limited to the following:
```
Comment (czoido) [inline creating_packages/toolchains/meson.rst:36]: I think the following properties are missing here

Comment (SSE4) [reply creating_packages/toolchains/meson.rst:36]: yeah, forget to stage changes, my fault 🤦‍♂️

File: creating_packages/toolchains/meson.rst [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+* Detection of ``default_library`` from Conan settings
+
+  * Based on existance/value of a option named ``shared``
+
+* Detection of ``buildtype`` from Conan settings
+
+* Definition of the C++ standard as necessary
+
+- The Visual Studio runtime (``b_vscrt ``), obtained from Conan input settings
```
Comment (czoido) [inline creating_packages/toolchains/meson.rst:46]: I think the formatting here is making the PR fail

**Review by czoido — APPROVED** [review]

**Pull request merged**